src/modules/ekf2/EKF/mag_3d_control.cpp

```diff
@@ -63,6 +63,7 @@ void Ekf::controlMag3DFusion(const magSample &mag_sample, const bool common_star
 	_control_status.flags.mag_3D = (_params.mag_fusion_type == MagFuseType::AUTO)
 				       && _control_status.flags.mag
 				       && _control_status.flags.mag_aligned_in_flight
+				       && _control_status.flags.mag_consistent
```

Comment:
Do we need this one or could we change it a bit? I might be overthinking it, but I was trying to minimize the intertwined dependencies between mag_hdg and mag_3d.

For example we call controlMag3DFusion() before controlMagHeadingFusion(), so it could be possible to have a single cycle of mag_3d active (during _yaw_angle_observable), but then immediately mag_hdg marks it inconsistent and on the next cycle mag_3d won't continue. The published estimator_status_flags will show both cs_mag_3d and !cs_mag_consistent. This particular case is probably pretty harmless, but it's the type thing I was hoping to eliminate in maintaining separate logic.

Reply:
The mag heading consistency is more a general "mag check", I think it makes sense to use it in all types of mag fusion that are not only affecting the mag states.